sklearnex/preview/linear_model/ridge.py

```diff
@@ -0,0 +1,375 @@
+                fit_intercept=True,
+                copy_X=True,
+                max_iter=None,
+                tol=1e-4,
+                solver="auto",
+                positive=False,
+                random_state=None,
+            ):
+                super().__init__(
+                    alpha=alpha,
+                    fit_intercept=fit_intercept,
+                    copy_X=copy_X,
+                    max_iter=max_iter,
+                    tol=tol,
+                    solver=solver,
+                    positive=positive,
+                    random_state=random_state,
+                )
+
+        else:
+
+            def __init__(
+                self,
+                alpha=1.0,
+                fit_intercept=True,
+                normalize="deprecated" if sklearn_check_version("1.0") else False,
+                copy_X=True,
+                max_iter=None,
+                tol=1e-4,
+                solver="auto",
+                positive=False,
+                random_state=None,
+            ):
+                super().__init__(
+                    alpha=alpha,
+                    fit_intercept=fit_intercept,
+                    normalize=normalize,
+                    copy_X=copy_X,
+                    max_iter=max_iter,
+                    solver=solver,
+                    tol=tol,
+                    positive=positive,
+                    random_state=random_state,
+                )
+
+        def fit(self, X, y, sample_weight=None):
+            # It is necessary to properly update coefs for predict if we
+            # fallback to sklearn in dispatch
+            if hasattr(self, "_onedal_estimator"):
+                del self._onedal_estimator
+
+            dispatch(
+                self,
+                "fit",
+                {
+                    "onedal": self.__class__._onedal_fit,
+                    "sklearn": sklearn_Ridge.fit,
+                },
+                X,
+                y,
+                sample_weight,
+            )
+            return self
+
+        @wrap_output_data
+        def predict(self, X):
+            check_is_fitted(self)
+
+            return dispatch(
+                self,
+                "predict",
+                {
+                    "onedal": self.__class__._onedal_predict,
+                    "sklearn": sklearn_Ridge.predict,
+                },
+                X,
+            )
+
+        @wrap_output_data
+        def score(self, X, y, sample_weight=None):
+            check_is_fitted(self)
+
+            return dispatch(
+                self,
+                "score",
+                {
+                    "onedal": self.__class__._onedal_score,
+                    "sklearn": sklearn_Ridge.score,
+                },
+                X,
+                y,
+                sample_weight=sample_weight,
+            )
+
+        def _onedal_fit_supported(self, patching_status, method_name, *data):
+            assert method_name == "fit"
```

Comment:
agree but it is following sklearnex/lin_reg style, so kept it

Reply:
What is this style and why is it needed here?

Reply:
my ideal understanding is: if i remove it here, it should be removed everywhere where it can be considered an "overkill", which is preview kmeans and sklearnex linear regression. changing it only here would increase variance and i believe methods with similar purpose across project should implicitly tend towards similar implementation/style; so i would either change it everywhere or not touch at all. let me know your thoughts on this

Reply:
Probably Linreg and KMeans are the only ones that have this style. My suggestion would be to refactor this on a separate PR.

Reply:
+1 for a design document, too many styles tbh